src/Middleware/StaticFiles/src/StaticFileContext.cs

```diff
@@ -78,56 +76,54 @@ public StaticFileContext(HttpContext context, StaticFileOptions options, PathStr
             _ifModifiedSinceState = PreconditionState.Unspecified;
             _ifUnmodifiedSinceState = PreconditionState.Unspecified;
             _range = null;
-            _isRangeRequest = false;
+            _requestType = RequestType.Unspecified;
         }
 
-        internal enum PreconditionState
-        {
-            Unspecified,
-            NotModified,
-            ShouldProcess,
-            PreconditionFailed
-        }
+        private RequestHeaders RequestHeaders => (_requestHeaders ??= _request.GetTypedHeaders());
 
-        public bool IsHeadMethod
-        {
-            get { return _isHead; }
-        }
+        private ResponseHeaders ResponseHeaders => (_responseHeaders ??= _response.GetTypedHeaders());
 
-        public bool IsRangeRequest
-        {
-            get { return _isRangeRequest; }
-        }
+        public bool IsHeadMethod => (_requestType & RequestType.IsHead) != 0;
 
-        public string SubPath
-        {
-            get { return _subPath.Value; }
-        }
+        public bool IsGetMethod => (_requestType & RequestType.IsGet) != 0;
 
-        public string PhysicalPath
-        {
-            get { return _fileInfo?.PhysicalPath; }
-        }
+        public bool IsRangeRequest => (_requestType & RequestType.IsRange) != 0;
```

Comment:
Add a private setter? Would prefer to keep the bitwise code isolated.

Reply:
Can use `HasFlag` instead; keep forgetting it should be fast now.

Reply:
What about the private setter?

Reply:
Done